tests/e2e/scale/test_cephfs_many_files.py

```diff
@@ -0,0 +1,195 @@
+TARFILE = 'cephfs.tar.gz'
+SIZE = '20Gi'
+TFILES = 1000000
+SAMPLE_TEXT = b"A"
+
+log = logging.getLogger(__name__)
+
+
+def add_million_files():
+    """
+    Create a directory with one million files in it.
+    Tar that directory to a zipped tar file.
+    rsynch that tar file to the cephfs pod
+    Extract the tar files on ceph pod onto the mounted ceph filesystem.
+
+    Returns:
+        list: list of ten of the files created.
+    """
+    logging.info(f"Creating {TFILES} files on Cephfs")
+    onetenth = TFILES / 10
+    endoften = onetenth - 1
+    ntar_loc = mkdtemp()
+    tarfile = os.path.join(ntar_loc, TARFILE)
+    new_dir = mkdtemp()
+    test_file_list = []
+    for i in range(0, TFILES):
+        tmpfile = NamedTemporaryFile(dir=new_dir, delete=False)
+        fname = tmpfile.name
+        with tmpfile:
+            tmpfile.write(SAMPLE_TEXT)
+        if i % onetenth == endoften:
+            dispv = i + 1
+            logging.info(f'{dispv} local files created')
+            test_file_list.append(fname.split(os.sep)[-1])
+    tmploc = ntar_loc.split('/')[-1]
+    run_cmd(f'tar cfz {tarfile} -C {new_dir} .')
+    ocp_obj = ocp.OCP(
+        kind=constants.POD,
+        namespace=constants.DEFAULT_NAMESPACE
+    )
+    ocp_obj.exec_oc_cmd(
+        f'rsync {ntar_loc} {POD_NAME}:{constants.MOUNT_POINT}',
+        timeout=300
+    )
+    ocp_obj.exec_oc_cmd(
+        f'exec {POD_NAME} -- mkdir {constants.MOUNT_POINT}/x'
+    )
+    ocp_obj.exec_oc_cmd(
+        f'exec {POD_NAME} -- /bin/tar xf'
+        f' {constants.MOUNT_POINT}/{tmploc}/{TARFILE}'
+        f' -C {constants.MOUNT_POINT}/x',
+        timeout=1800
+    )
+    rmtree(new_dir)
+    os.remove(tarfile)
+    return test_file_list
+
+
+class MillionFilesOnCephfs(object):
+    """
+    Create pvc and cephfs pod, make sure that the pod is running.
+    """
+    def __init__(self):
+        self.cephfs_pvc = helpers.create_pvc(
+            constants.DEFAULT_STORAGECLASS_CEPHFS,
+            pvc_name=PVC_NAME,
+            namespace=constants.DEFAULT_NAMESPACE,
+            size=SIZE
+        )
+        helpers.wait_for_resource_state(
+            self.cephfs_pvc,
+            constants.STATUS_BOUND,
+            timeout=300
+        )
+        self.cephfs_pod = helpers.create_pod(
+            interface_type=constants.CEPHFILESYSTEM,
+            pvc_name=self.cephfs_pvc.name,
+            namespace=constants.DEFAULT_NAMESPACE,
+            node_name='compute-0',
+            pod_name=POD_NAME
+        )
+        helpers.wait_for_resource_state(
+            self.cephfs_pod,
+            constants.STATUS_RUNNING,
+            timeout=300
+        )
+        logging.info("pvc and cephfs pod created")
+        self.test_file_list = add_million_files()
+        logging.info("cephfs test files created")
+
+    def cleanup(self):
+        self.cephfs_pod.delete()
+        self.cephfs_pvc.delete()
+        logging.info("Teardown complete")
+
+
+@pytest.fixture(scope='session')
```

Comment:
@wusui Please change the fixture scope to "class", if we keep it as session then in complete "scale" suite execution fixture will not be teardown till all the test executions completed.